.github/scripts/checkKnipRegression.ts

```diff
@@ -25,31 +29,43 @@ function compareKnipResults(
   const rows: Array<Record<string, string | number>> = []
   for (const [category, baseProblems] of Object.entries(baseKnipResults)) {
     const branchProblems = branchKnipResults[category] ?? 0
+    const hasRegression = branchProblems > baseProblems
     rows.push({
-      'Issue Category': category.trim(),
-      'Base Branch': baseProblems,
-      'PR Branch': branchProblems,
+      'Issue Category': hasRegression ? chalk.red(category.trim()) : category.trim(),
+      'Base Branch': hasRegression ? chalk.green(baseProblems) : baseProblems,
+      'PR Branch': hasRegression ? chalk.red(branchProblems) : branchProblems,
     })
-    if (branchProblems > baseProblems) {
+    if (hasRegression) {
       shouldFail = true
     }
   }
-  console.table(rows)
+
+  // Use easy-table instead of console.table, because console.table doesn't support colors (it escapes them)
+  console.log(easyTable.print(rows))
   return shouldFail
 }
 
 if (process.env.GITHUB_EVENT_NAME === 'pull_request') {
-  const branchKnipOutput = $.exec('yarn knip --no-gitignore').stdout.trim()
+  const branchKnipOutput = stripAnsi($.exec('yarn knip --no-gitignore').stdout.trim())
   const branchKnipResults = parseKnipOutput(branchKnipOutput)
 
   // checkout pr base. More info here: https://github.com/actions/checkout/tree/v3?tab=readme-ov-file#checkout-v3
   $.exec('git checkout HEAD^')
 
-  const baseKnipOutput = $.exec('yarn knip --no-gitignore').stdout.trim()
+  const baseKnipOutput = stripAnsi($.exec('yarn knip --no-gitignore').stdout.trim())
   const baseKnipResults = parseKnipOutput(baseKnipOutput)
 
   if (compareKnipResults(baseKnipResults, branchKnipResults)) {
     console.log('Knip check failed. PR branch reported more problems than base branch.')
+
+    // Print diff of knip output, for easily seeing what changed
+    console.log(
+      diff(baseKnipOutput, branchKnipOutput, {
+        // limit the number of common lines to print
+        expand: false,
+      })
+    )
```

Comment:
I think this a good enough for our purposes for now but wanted to flag a couple of things:

1. For some categories, the output includes whitespaces based on the length of the filename and unused exported item, so if the item with max length changes between the base and PR branches, we would have all lines within the category reported as a diff. One option could be to strip contiguous whitespaces for those categories.
2. If a PR removes some unused code in one category, but adds in another category, the diff would include the former as well. Ideally we'd want only the latter reported in the diff. 

I attempted using the JSON format for knip output and get a set difference, which should address the above, but for some reason the output was truncated by shelljs.

Reply:
Oh that’s super helpful!
Is this the shelljs bug you ran into?
https://github.com/shelljs/shelljs/issues/1005

maybe we could try the suggestion? or redirect the output to a temp file we would read afterward?

Reply:
It looks similar yes, but I was seeing different truncations in my local and CI, here's a CI run with truncation https://github.com/valora-inc/wallet/actions/runs/7295837665/job/19882780114. I recall trying both maxBuffer and redirecting the output to a file and saw the same result, but haven't tried on CI